backend/compact-connect/lambdas/python/common/cc_common/data_model/data_client.py

```diff
@@ -701,28 +712,28 @@ def create_military_affiliation(
         schema = MilitaryAffiliationRecordSchema()
         latest_military_affiliation_record_serialized = schema.dump(latest_military_affiliation_record)
 
-        with self.config.provider_table.batch_writer() as batch:
-            batch.put_item(Item=latest_military_affiliation_record_serialized)
-
-        # We need to check for any other military affiliations with an 'active' status for this provider
+        # We need to check for any other military affiliations for this provider
         # and set them to inactive. Note these could be consolidated into a single batch call if performance
         # becomes an issue.
         self.inactivate_military_affiliation_status(compact, provider_id)
 
+        with self.config.provider_table.batch_writer() as batch:
+            batch.put_item(Item=latest_military_affiliation_record_serialized)
+
         return latest_military_affiliation_record
 
     def inactivate_military_affiliation_status(self, compact: str, provider_id: str):
         """
-        Sets all active military affiliation records to an inactive status for a provider in the database.
+        Sets all military affiliation records to an inactive status for a provider in the database.
 
         :param compact: The compact name
         :param provider_id: The provider id
         :return: None
         """
-        active_military_affiliation_records = self._get_active_military_affiliation_records(compact, provider_id)
+        military_affiliation_records = self._get_all_military_affiliation_records_for_provider(compact, provider_id)
         schema = MilitaryAffiliationRecordSchema()
         with self.config.provider_table.batch_writer() as batch:
-            for record in active_military_affiliation_records:
+            for record in military_affiliation_records:
                 record['status'] = MilitaryAffiliationStatus.INACTIVE.value
                 serialized_record = schema.dump(record)
                 batch.put_item(Item=serialized_record)
```

Comment:
Now here, you would use the data class to set the status value using the update method, and then serializing it into a database record:
```suggestion
                record.update({'status': MilitaryAffiliationStatus.INACTIVE.value})
                serialized_record = record.serialize_to_database_record()
                batch.put_item(Item=serialized_record)
```